marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/entity-lib.sjs

```diff
@@ -370,8 +370,11 @@ function validateModelDefinitions(definitions) {
     }
     if (definitions[entityName].properties) {
       Object.keys(definitions[entityName].properties).forEach(propertyName => {
-        if (!pattern.test(propertyName)) {
-          throw new Error(`Invalid property name: ${propertyName}; must start with a letter and can only contain letters, numbers, hyphens, and underscores.`);
+        try{
+          fn.QName('',propertyName)
+        }
+        catch(ex){
+          throw new Error(`Invalid property name: ${propertyName} in entity model ${entityName}; It must be a valid NCName.`);
```

Comment:
I think we need a little more info here, because I don't anticipate that Stewart the modeler will have ever heard of "NCName" before. 

How about: "; it must be a valid NCName as defined at http://www.datypic.com/sc/xsd/t-xsd_Name.html" (that page provides a nice description at the top; I'm open to including that description too - i.e. must start with a letter, underscore(_), or colon (:), and may contain only letters, digits, underscores (_), colons (:), hyphens (-), and periods (.) "